### PR TITLE
Added tests for linked_list.py #39

### LINKED ISSUE #39: Needed test file for existing linked_list.py
- [ ] Using unittest framework, write test cases for linked_list.py

### PR DESCRIPTION
**Pull request by Aryamanz29**
-Added test file for linked_list.py
-remove linked_list_tests.ipynb from dsalgo/tests
closes #39 
Screenshot (65)

### CODE REVIEW

**Review by codesankalp — APPROVED** [review]

File: tests/linkedlist_tests.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  
+    def test_toLinkedlist(self):
+        self.list=[1,2,5,6,7] 
+        self.ls.to_linked_list([6,7])
+        self.assertEqual(self.list,self.ls.to_list())
+
+    def test_insert(self):
+          self.list=[1,2,5,4]
+          self.ls.insert(4,12)
+          self.assertEqual(self.list,self.ls.to_list())
```
Comment (codesankalp) [inline tests/linkedlist_tests.py:32]: use flake8 for styling purpose.

**Review by codesankalp — CHANGES_REQUESTED** [review]

File: tests/linkedlist_tests.py (hunk 1 above)
Comment (Aryamanz29) [reply tests/linkedlist_tests.py:32]: already checked pep8 styling no issue

File: tests/linkedlist_tests.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    def test_swapNodes(self):
+        self.list=[1,5,2]
+        self.ls.swap_nodes(1,2)
+        self.assertEqual(self.list,self.ls.to_list())
+ 
+if __name__=='__main__':
+    unittest.main()
+
+
+
```
Comment (devkapilbansal) [inline tests/linkedlist_tests.py:47]: Why so many blank lines here?

Comment (Aryamanz29) [reply tests/linkedlist_tests.py]: Done 😊

**Review by ravi5175 — APPROVED** [review]

File: tests/linkedlist_tests.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import unittest
```
Comment (17sushmita) [inline tests/linkedlist_tests.py:1]: rename this file to test_linkedlist.py

File: tests/linkedlist_tests.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        self.assertEqual(self.list, self.ls.to_list())
+
+    def test_swapNodes(self):
+        self.list = [1, 5, 2]
+        self.ls.swap_nodes(1, 2)
+        self.assertEqual(self.list, self.ls.to_list())
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (17sushmita) [inline tests/linkedlist_tests.py:47]: remove the driver code

File: tests/linkedlist_tests.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+from dsalgo.linked_list import LinkedList
+
+
+class TestList(unittest.TestCase):
+    def setUp(self):
+        self.ls = LinkedList()
+        self.ls.append(1)
+        self.ls.append(2)
+        self.ls.append(5)
+
```
Comment (17sushmita) [inline tests/linkedlist_tests.py:11]: for good testing practice, you can use random module and generate a list of random numbers to test

**Review by 17sushmita — CHANGES_REQUESTED** [review]

**Comment by Aryamanz29** [comment]
One thing I noticed when writing the test for the Linked list is that reverse() is not working properly, It only reverses the list when sorted & not working for random values. So please check linked_list.py if you got the same problem, we can create a new issue for that including the test.

**Comment by 17sushmita** [comment]
> One thing I noticed when writing the test for the Linked list is that reverse() is not working properly, It only reverses the list when sorted & not working for random values. So please check linked_list.py if you got the same problem, we can create a new issue for that including the test.

Can you provide some screenshots because It is working for the test case I tested on.

**Comment by Aryamanz29** [comment]
> > One thing I noticed when writing the test for the Linked list is that reverse() is not working properly, It only reverses the list when sorted & not working for random values. So please check linked_list.py if you got the same problem, we can create a new issue for that including the test.
> 
> Can you provide some screenshots because It is working for the test case I tested on.

Screenshot (70)
Screenshot (69)

**Comment by codesankalp** [comment]
@Aryamanz29 there is no problem in the code actually you tried the wrong way.
see this:
```
self.rev1 = self.list.copy()[ : :-1]
self.ls.reverse()
self.assertEqual(self.rev1, self.ls.to_list())
```
I think with the above code you will understand what is wrong.

**Comment by Aryamanz29** [comment]
> @Aryamanz29 there is no problem in the code actually you tried the wrong way.
> see this:
> 
> ```
> self.rev1 = self.list.copy()[ : :-1]
> self.ls.reverse()
> self.assertEqual(self.rev1, self.ls.to_list())
> ```
> 
> I think with the above code you will understand what is wrong.

ohk i got this thanks👍

**Pull request closed without merging**